src/runner/test-run-controller.ts

```diff
@@ -69,6 +83,13 @@ class Quarantine {
 
         return { failedTimes, passedTimes };
     }
+
+    private _getFailedThreshold (): number {
+        if (this.TEST_RUN_THRESHOLD !== DEFAULT_TEST_RUN_THRESHOLD)
+            return this.TEST_RUN_THRESHOLD - this.PASSED_QUARANTINE_THRESHOLD + 1;
+
+        return DEFAULT_QUARANTINE_THRESHOLD;
+    }
```

Comment:
you can make the new property from the `_getFailedThreshold` function as well. It's up to you